horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1013,181 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
       DatasetList list = new DatasetList();
       return list;
    }
+
+   private void transform(Run run, String role, String token) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+
+      Collection<RunService.RunSchema> runSchemas = getSchemas(run, role, token);
+
+      try (@SuppressWarnings("unused") CloseMe h1 = roleManager.withRoles(em, role);
+            @SuppressWarnings("unused") CloseMe h2 = roleManager.withToken(em, token)) {
+
+         Test test = Test.findById(run.testid);
+         boolean matched = false;
+         Map<Transformer, List<JsonNode>> transformedNodeResults = new HashMap<>();
+         Set<Transformer> order = new TreeSet<>();
+         Collection<DataSet> dataSets = new ArrayList<>();
+         int ordinal = 0;
+         boolean functionError = false;
+
+         Collection<Transformer> transformers = test.transformers;
+
+         boolean containsArray = false;
+
+         for ( RunService.RunSchema s : runSchemas ) {
+
+            for (Transformer t : transformers) {
```

Comment:
>  We need to define what happens next as Horreum will be rather inflexible about that: Either we could concatenate the results in the intermediary representation (before merging results of transformers) so that each item in the results ends up in its own DataSet. Alternatively we could treat those as results from two unrelated Transformers and add those side-by-side to single dataset.
 I am inclined to put a copy of each scheam (without Transformers) in each DataSet.

Reply:
For now I have treated a schema in the same way as Transformers. To conform with the same behaviour when it comes to merging. If this causes problems as it might then we can switch.